eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+    & git config user.name $githubUser
+    Pop-Location
+}
+
+function Get-Repo-Location($repoName)
+{
+    "$testRoot\$repoName"
+}
+
+function Git-Command($repoName) {
+    Push-Location -Path $(Get-Repo-Location($repoName))
+    try {
+        $gitParams = $args
+        if ($gitParams.GetType().Name -ne "Object[]") {
+            $gitParams = $gitParams.ToString().Split(" ")
+        }
+        Write-Host "Running 'git $gitParams' from $(Get-Location)"
+        $commandOutput = & git @gitParams; if ($LASTEXITCODE -ne 0) { throw "Git exited with exit code: $LASTEXITCODE" } else { $commandOutput }
+        $commandOutput
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+## If able to retrieve a build, get the SHA that it was built from
+$sha = Get-LastKnownGoodBuildSha -azdoOrg $azdoOrg -azdoProject $azdoProject -buildDefinitionId $buildDefinitionId -azdoToken $azdoToken
+
+## Clone the repo from git
+GitHub-Clone $targetRepoName
+
+## Create a branch from the repo with the given SHA. 
+$targetBranch = "dev/" + $githubUser + "/arcade-" + $arcadeSdkVersion
+$githubUri = Get-Github-RepoAuthUri $targetRepoName
+
+$darcBranchName = "refs/heads/" + $targetBranch
+$darcRepoName = "https://github.com/${githubOrg}/${targetRepoName}"
+
+# check to see if branch exists and clean it up if it does
+$branchExists = Git-Command $targetRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+if($null -ne $branchExists)
+{
+    & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+    Git-Command $targetRepoName push origin --delete $targetBranch
+}
+Git-Command $targetRepoName checkout -b $targetBranch $sha
+
+## Make the changes to that branch to update Arcade - use darc
+Set-Location $(Get-Repo-Location $targetRepoName)
+& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --password $barToken
+
+## Push branch to github
+Git-Command $targetRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
+Git-Command $targetRepoName push origin HEAD
+
+## Push branch to AzDO org/project with the official pipeline to build the repo
+## Don't need to do this for Roslyn, but we'll need to do it for Runtime. 
+
+## Add default channel from that github repo and branch to "General Testing"
+& darc add-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+
+## Run an official build of the branch using the official pipeline
+$buildId = Invoke-AzDOBuild -azdoOrg $azdoOrg -azdoProject $azdoProject -buildDefinitionId $buildDefinitionId -azdoToken $azdoToken -branchName $targetBranch -buildParameters $buildParameters
+
+## Check build for completion every 5 minutes. 
+while("completed" -ne (Get-BuildStatus -azdoOrg $azdoOrg -azdoProject $azdoProject -buildId $buildId -azdoToken $azdoToken))
```

Comment:
Would be nice to print the link to the build at least once so that we can easily click + inspect if needed.

Reply:
Good idea :)  I added it.